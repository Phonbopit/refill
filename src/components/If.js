import React from 'react'

export const If = ({ when, children }) => when ? children : null

