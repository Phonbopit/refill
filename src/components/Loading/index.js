import React from 'react'

export default () => (
  <div className="container">
    <p>Loading ...</p>
  </div>
)
