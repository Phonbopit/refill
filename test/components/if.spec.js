import React from 'react'
import renderer from 'react-test-renderer'

import { If } from '../../src/components/If'

it('<If> should render when true', () => {
  const component = (
    <If render={true}>
      <h1>You can see me :)</h1>
    </If>
  )

  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

it('<If> should not show when false', () => {
  const component = (
    <If render={false}>
      <h1>You can't see me :)</h1>
    </If>
  )

  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
