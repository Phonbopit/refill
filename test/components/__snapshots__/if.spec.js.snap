// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<If> should not show when false 1`] = `null`;

exports[`<If> should render when true 1`] = `
<h1>
  You can see me :)
</h1>
`;
